app/controllers/save_pages_controller.rb

```diff
@@ -0,0 +1,52 @@
+class SavePagesController < ApplicationController
+  include AccountConcern
+
+  before_action :set_no_cache_headers
+  before_action { head :not_found unless feature_flag_enabled? }
+
+  def create
+    GdsApi.account_api.save_page(
+      page_path: page_path,
+      govuk_account_session: account_session_header,
+    )
+    redirect_to page_path + "?personalisation=page_saved"
+  rescue GdsApi::HTTPUnauthorized
+    authenticate(save_page_path(page_path: page_path))
+  rescue GdsApi::HTTPUnprocessableEntity
+    redirect_to page_path + "?personalisation=page_not_saved"
+  end
+
+  def destroy
+    GdsApi.account_api.delete_saved_page(
+      page_path: page_path,
+      govuk_account_session: account_session_header,
+    )
```

Comment:
This can also throw a 404 error, which we should treat as success, because it means the page has already been deleted (or wasn't saved)

Reply:
we could reuse `personalisation=page_removed`
or do `personalisation=page_already_removed`

And let the frontend handle that?

Reply:
Keep it simple for now, go for `page_removed` until we have a need to treat the two cases differently